Client/U2FExtensions.swift

```diff
@@ -0,0 +1,893 @@
+    }
+    
+    private func handleFIDORegistration(handle: Int, request: FIDORegisterRequest, requestId: Int) {
+        guard let registerRequest = YKFKeyU2FRegisterRequest(challenge: request.challenge, appId: request.appId ?? "") else {
+            sendFIDORegistrationError(handle: handle, requestId: requestId, errorCode: U2FErrorCodes.bad_request)
+            return
+        }
+        
+        guard let u2fservice = YubiKitManager.shared.keySession.u2fService else {
+            sendFIDORegistrationError(handle: handle, requestId: requestId, errorCode: U2FErrorCodes.other_error)
+            return
+        }
+        
+        u2fservice.execute(registerRequest) { [weak self] response, error in
+            guard let self = self else {
+                log.error(U2FErrorMessages.Error.rawValue)
+                return
+            }
+            
+            guard error == nil else {
+                let errorMessage = error?.localizedDescription ?? Strings.U2FRegistrationError
+                self.sendFIDORegistrationError(handle: handle, requestId: requestId, errorCode: U2FErrorCodes.other_error, errorMessage: errorMessage)
+                return
+            }
+            guard let clientData = response?.clientData.websafeBase64String() else {
+                self.sendFIDORegistrationError(handle: handle, requestId: requestId, errorCode: U2FErrorCodes.other_error)
+                return
+            }
+            guard let registrationData = response?.registrationData.websafeBase64String() else {
+                self.sendFIDORegistrationError(handle: handle, requestId: requestId, errorCode: U2FErrorCodes.other_error)
+                return
+            }
+            
+            self.cleanupFIDORegistration(handle: handle)
+            if requestId > -1 {
+                ensureMainThread {
+                    self.tab?.webView?.evaluateJavaScript("u2f.postLowLevelRegister(\(requestId), \(true), '\(request.version)', '\(registrationData)', '\(clientData)', \(defaultErrorCode), '')", completionHandler: { _, error in
+                        if error != nil {
+                            let errorDescription = error?.localizedDescription ?? U2FErrorMessages.ErrorRegistration.rawValue
+                            log.error(errorDescription)
+                        }
+                    return
+                }) }
+                return
+            }
+
+            ensureMainThread {
+                self.tab?.webView?.evaluateJavaScript("u2f.postRegister('\(handle)', \(true), '\(request.version)', '\(registrationData)', '\(clientData)', \(defaultErrorCode), '')", completionHandler: { _, error in
+                    if error != nil {
+                        let errorDescription = error?.localizedDescription ?? U2FErrorMessages.ErrorRegistration.rawValue
+                        log.error(errorDescription)
+                    }
+                return
```

Comment:
Im just going to stop here and say there a lots of redundant returns below that should be removed :)